radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/TimestampedECDSASignatures.java

```diff
@@ -108,10 +109,19 @@
 	@DsonOutput(DsonOutput.Output.ALL)
 	private SerializerDummy serializer = SerializerDummy.DUMMY;
 
-	private Map<BFTNode, TimestampedECDSASignature> nodeToTimestampedSignature;
+	private final Map<BFTNode, TimestampedECDSASignature> nodeToTimestampedSignature;
 
 	@JsonCreator
-	public static TimestampedECDSASignatures from(@JsonProperty("signatures") Map<String, TimestampedECDSASignature> signatures) {
+	public static TimestampedECDSASignatures from(
+		@JsonProperty("signatures") Map<String, TimestampedECDSASignature> signatures
+	) {
+		if (signatures != null) {
+			signatures.forEach((key, value) -> {
```

Comment:
Maybe we should add similar check in the constructor.

Reply:
added